pkg/referrerstore/oras/oras.go

```diff
@@ -117,30 +126,33 @@ func (store *orasStore) GetConfig() *config.StoreConfig {
 	return &store.rawConfig
 }
 
-func (store *orasStore) ListReferrers(ctx context.Context, subjectReference common.Reference, artifactTypes []string, nextToken string) (referrerstore.ListReferrersResult, error) {
-	// TODO: handle nextToken
-	registryClient, err := store.createRegistryClient(ctx, subjectReference)
+func (store *orasStore) ListReferrers(ctx context.Context, subjectReference common.Reference, artifactTypes []string, nextToken string, subjectDesc ...*ocispecs.SubjectDescriptor) (referrerstore.ListReferrersResult, error) {
+	repository, err := store.createRepository(ctx, subjectReference)
 	if err != nil {
 		return referrerstore.ListReferrersResult{}, err
 	}
 
-	ref := fmt.Sprintf("%s@%s", subjectReference.Path, subjectReference.Digest)
-	var referrerDescriptors []artifactspec.Descriptor
-	if artifactTypes == nil {
-		artifactTypes = []string{""}
-	}
-	for _, artifactType := range artifactTypes {
-		_, res, err := oras.Discover(ctx, registryClient.Resolver, ref, artifactType)
+	// resolve subject descriptor if not provided
+	var resolvedSubjectDesc *ocispecs.SubjectDescriptor
+	if len(subjectDesc) > 0 {
+		resolvedSubjectDesc = subjectDesc[0]
```

Comment:
Curious as to why do we need an array for the subject descriptors? Wouldn't a reference resolve to a descriptor?

Reply:
Yes you're correct. I was trying to mimic an optional parameter by creating the parameter as a variable argument. I don't believe Go supports optional parameters?

Reply:
If all callers do pass this parameter as of current, do you think we can make it a normal parameter (and have default as nil)

Reply:
Sure sounds good. I switched over to making it a normal parameter